config_option_database/utils/Yacc2Graph.py

```diff
@@ -0,0 +1,47 @@
+#############################################################################
+# Copyright (c) 2019 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+from subprocess import DEVNULL, Popen
+from tempfile import NamedTemporaryFile
+
+
+def _run_in_shell(command):
+    proc = Popen(command, stderr=DEVNULL, stdout=DEVNULL)
+    proc.wait()
+    return proc.returncode == 0
+
+def _write_to_file(string):
+    file = NamedTemporaryFile(mode='w')
+    file.write(string)
+    file.flush()
+    file.seek(0)
+    return file
+
+def _yacc2xml(yacc):
+    with _write_to_file(yacc) as file:
+        output = '/tmp/yacc2xml.xml'
```

Comment:
`bison` takes a `.y` file (NamedTemporaryFile from `_write_to_file()`) and outputs a `.xml` file, which is our output.

Reply:
Oh ok I see.

Why do you use temporary file for yacc, but hardcoded file for the xml output?

Reply:
I will change it to `output = NamedTemporaryFile().name`. It will close and delete the file, but generate a valid filename under `/tmp/`. After this, `bison` will generate the xml file at that location, and later we will delete that in `_yacc2rules()`.

Reply:
Done!